indexer-balances/src/db_adapters/balance_changes.rs

```diff
@@ -699,3 +743,62 @@ async fn get_account_view(
         ),
     }
 }
+
+// Feature "rpc-sanity-check" performs an extra call to the RPC (RPC_SANITY_CHECK_URL)
+// compares the results and falls back to the RPC_SANITY_CHECK_URL's result in case of mismatch
+// The corresponding warning log is being emitted.
+#[cfg(feature = "rpc-sanity-check")]
+async fn rpc_sanity_check(
+    sanity_json_rpc_client: &near_jsonrpc_client::JsonRpcClient,
+    account_id: &near_indexer_primitives::types::AccountId,
+    block_hash: &near_indexer_primitives::CryptoHash,
+    account_response: near_jsonrpc_primitives::types::query::RpcQueryResponse,
+) -> near_jsonrpc_primitives::types::query::RpcQueryResponse {
+    let query = near_jsonrpc_client::methods::query::RpcQueryRequest {
+        block_reference: near_primitives::types::BlockReference::BlockId(
+            near_primitives::types::BlockId::Hash(*block_hash),
+        ),
+        request: near_primitives::views::QueryRequest::ViewAccount {
+            account_id: account_id.clone(),
+        },
+    };
+
+    let sanity_check_account_response = match sanity_json_rpc_client.call(query).await {
+        Ok(res) => res,
+        Err(err) => {
+            tracing::error!(
+                target: crate::LOGGING_PREFIX,
+                "RPC sanity check call failed with error \n{:#?}\n
+                Returning the result from RPC_URL. Data sanity might be corrupted!",
+                err,
+            );
+            return account_response;
+        }
+    };
+
+    // We have to unpack the AccountView in order to compare the results
+    // Since near_jsonrpc_primitives::types::query::RpcQueryResponse doesn't implement Eq
+    // While it is an extra work it should be fine since we're not going to run it in production
```

Comment:
So we don't plan to merge it, right? We will only use this branch for dev

Reply:
I just realized it's a feature
I'm OK with merging this if we continue using RPC_URL for stable RPC url

Reply:
Let me clarify my initial intention.

I wanted us to use `RPC_URL` as the main source of data for the indexer. Meaning we use it to query the RPC service (doesn't matter reliable or experimental).

The feature is supposed to be used when we have some doubts about the source we're going to use and we do want to add a sanity check.

That's why I wanted `RPC_SANITY_CHECK_URL` to be the reliable one.

Once experiments are done and we're good to go, we just drop the feature and keep all the configs as is.

Does it sound OK to you? 

Option 1. I change the logic for RPC_URL to be stable/reliable and change RPC_SANITY... to experimental-something
Option 2. I fix doc-strings and remove the confusion I introduced